### PR TITLE
[language] Making gas counters more usable

### PR DESCRIPTION
**Pull request by tzakian**
Updates the counters somewhat to make them a bit nicer to use.

### CODE REVIEW

**Comment by tzakian** [comment]
This need to be reworked; per-transaction based stats don't work with prometheus since it's a time-series database. We'll need to instead do something else to correlate these two values.

File: language/vm/vm_runtime/src/counters.rs [hunk 1]
```diff
@@ -16,52 +16,60 @@ const TXN_EXECUTION_DISCARD: &str = "txn.execution.discard";
...
 macro_rules! record_stats {
-    ($e:expr) => {
-        $e
+    // Gather some information that is only needed in relation to recording statistics
+    (info | $($stmt:stmt);+;) => {
+        $($stmt);+;
     };
-    ($histrogram:expr, $block:block) => {{
+    // Per-transaction info: Set the gas histogram for $ident to be $amount.
+    (gas | $ident:ident | $amount:expr) => {
```
Comment (runtian-zhou) [inline language/vm/vm_runtime/src/counters.rs]: IMHO we can use this macro to record all integer statistics. Should we change the name to make it sounds a bit more generic?

Comment (tzakian) [reply language/vm/vm_runtime/src/counters.rs]: Good idea. I've updated the macro to generalize this, and allow us to also set other types of statistics as well.

**Pull request closed without merging**